Clean up Taxonomy API files/hooks/queries [FC-0036]

bradenmacdonald opened the pull request:
1. Addresses the TODOs in the existing taxonomy API code.
2. Consolidates the files in taxonomy/foo/data/api.* into just one taxonomy/foo/data/api.*
3. Incorporates some react-query best practices, like query key patterns - see  https://github.com/openedx/frontend-app-admin-portal/blob/2ba315d/docs/decisions/0006-tanstack-react-query.rst


Private ref: [FAL-3676](https://tasks.opencraft.com/browse/FAL-3676)

openedx-webhooks commented:
Thanks for the pull request, @bradenmacdonald! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.




<!-- comment:external_pr -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

bradenmacdonald commented on src/taxonomy/data/apiHooks.js:
```diff
@@ -12,95 +12,178 @@
+        const { data } = await getAuthenticatedHttpClient().put(apiUrls.tagsImport(taxonomyId), formData);
+        return camelCaseObject(data);
+      } catch (/** @type {any} */ err) {
+        throw new Error(err.response?.data.error || err.message);
```
The previous version of this (in `api.py`) had `err.response?.data` but I think that's wrong. I changed it to `err.response?.data.error`, which is the same as the "import plan" endpoint had.

ChrisChV commented on src/content-tags-drawer/ContentTagsDrawer.test.jsx:
```diff
@@ -30,18 +37,29 @@ jest.mock('./data/apiHooks', () => ({
+  });
+
   it('should render page and page title correctly', () => {
     const { getByText } = render(<RootWrapper />);
```
@bradenmacdonald Maybe we should also create a task to refactor the code following [good practices](https://forum.opencraft.com/t/some-tips-and-tricks-for-testing-react-code/1657)

ChrisChV commented:
@bradenmacdonald When importing a taxonomy, the tags are not updated in real time. This does work in `master`

https://www.loom.com/share/9bd7e8e1daea4d35b22cbeeff4943571?sid=1fef07f2-f4da-4ff6-88d5-23894b172812

ChrisChV reviewed (approved):
@bradenmacdonald Thanks for the work! Conditional approve, missing to solve https://github.com/openedx/frontend-app-course-authoring/pull/850#issuecomment-1976911074
- [X] I tested this: I tested all te pages and features of taxonomy
- [X] I read through the code and considered the security, stability and performance implications of the changes.
- [X] Includes tests for bugfixes and/or features added.
- [X] Includes documentation

bradenmacdonald commented:
@ChrisChV Sorry for the long delay here! And thanks for catching that issue. I have fixed the bug now :)
The problem was that I hadn't updated another apiHooks file to use the new query keys.
I also updated this with the latest commits from `master`.

Could you please look again?

ChrisChV reviewed (approved):
@bradenmacdonald Looks good :+1:

xitij2000 reviewed (commented):
Looks good! Will just test it once to see if everything is OK and then I think it's good to merge.

bradenmacdonald commented on src/taxonomy/data/api.js:
```diff
@@ -3,71 +3,125 @@ import { camelCaseObject, getConfig } from '@edx/frontend-platform';
+/**
+ * @param {string} path The subpath within the taxonomies "v1" REST API namespace
+ */
+const makeUrl = (path) => new URL(path, getTaxonomiesV1Endpoint());
```
Thanks, I like that more :)
Incorporated.

bradenmacdonald commented:
Made that change and updated to solve the conflict with master.

openedx-webhooks commented:
@bradenmacdonald 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/frontend-app-course-authoring&entry.867055334=https://github.com/openedx/frontend-app-course-authoring/pull/850&entry.1484655318=https://github.com/bradenmacdonald&entry.752974735=2024-02-22+02:17&entry.1917517419=2024-03-20+04:01&entry.2133058324=Yes) so we can improve your experience in the future.

<!-- comment:end_survey -->
